inference-engine/src/preprocessing/ie_preprocess_gapi.cpp

```diff
@@ -169,9 +169,50 @@ std::vector<std::vector<cv::gapi::own::Mat>> bind_to_blob(const I420Blob::Ptr& i
     std::vector<std::vector<cv::gapi::own::Mat>> batched_input_plane_mats(batch_size);
     for (int i = 0; i < batch_size; ++i) {
         auto& input = batched_input_plane_mats[i];
-        input.emplace_back(std::move(batched_y_plane_mats[i][0]));
-        input.emplace_back(std::move(batched_u_plane_mats[i][0]));
-        input.emplace_back(std::move(batched_v_plane_mats[i][0]));
+        input.emplace_back(std::move(batched_y_plane_mats[i][0])); //[0] cause only one channel for y
+        input.emplace_back(std::move(batched_u_plane_mats[i][0])); //[0] cause only one channel for u
+        input.emplace_back(std::move(batched_v_plane_mats[i][0])); //[0] cause only one channel for v
+    }
+
+    return batched_input_plane_mats;
+}
+
+std::vector<std::vector<cv::gapi::own::Mat>> bind_to_blob(const BatchedBlob::Ptr& blob,
+                                                          int batch_size) {
+    int underlied_blob_size = 1;
+
+    std::vector<std::vector<cv::gapi::own::Mat>> batched_input_plane_mats(blob->size() * underlied_blob_size);
+    for (size_t i = 0; i < blob->size(); ++i) {
+        auto inBlob = blob->getBlob(i);
+
+        //TODO reuse `underlied_blob_plane` memory
+        std::vector<std::vector<cv::gapi::own::Mat>> underlied_blob_plane;
+
+        if (inBlob->is<NV12Blob>()) {
+            underlied_blob_plane = bind_to_blob(as<NV12Blob>(inBlob), underlied_blob_size);
+        } else if (inBlob->is<I420Blob>()) {
+            underlied_blob_plane = bind_to_blob(as<I420Blob>(inBlob), underlied_blob_size);
+        } else if (inBlob->is<MemoryBlob>()) {
+            //correct batch_size for mem blob
+            std::shared_ptr<MemoryBlob> memBlob = as<MemoryBlob>(inBlob);
+            underlied_blob_size = static_cast<int>(memBlob->getTensorDesc().getDims()[0]);
```

Comment:
should some special extra intelligent method be called here to find `N` ?

Reply:
i didn't find it across dynamyc_cast and getDims request